From db9c0c4ab770bf34c687172ffb7ec39b68f98d42 Mon Sep 17 00:00:00 2001
From: Xu Jianqun <jay.xu@rock-chips.com>
Date: Wed, 3 Feb 2016 11:06:57 +0800
Subject: [PATCH] ARM64: dts: rockchip: add rk3368 sound node

Add sound node, use simple card driver for rt5640 codec on rockchip
rk3368 sheep board.

Change-Id: I1e7f0f7ca974d7232a2660219bd3ddfb408aaec3
Signed-off-by: Xu Jianqun <jay.xu@rock-chips.com>
---
 .../boot/dts/rockchip/rk3368-tb-sheep.dts     |  4 ++
 arch/arm64/boot/dts/rockchip/rk3368-tb.dtsi   | 42 +++++++++++++++++++
 2 files changed, 46 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3368-tb-sheep.dts b/arch/arm64/boot/dts/rockchip/rk3368-tb-sheep.dts
index 2e27c9c92098..ffd3a1e81f0c 100644
--- a/arch/arm64/boot/dts/rockchip/rk3368-tb-sheep.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3368-tb-sheep.dts
@@ -48,6 +48,10 @@
 	compatible = "rockchip,sheep", "rockchip,rk3368";
 };
 
+&rt5640 {
+	status = "okay";
+};
+
 &gt9xx {
 	status = "okay";
 };
diff --git a/arch/arm64/boot/dts/rockchip/rk3368-tb.dtsi b/arch/arm64/boot/dts/rockchip/rk3368-tb.dtsi
index 0010dc0fcb71..43f3360f414a 100644
--- a/arch/arm64/boot/dts/rockchip/rk3368-tb.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3368-tb.dtsi
@@ -63,6 +63,29 @@
 		};
 	};
 
+	sound {
+		compatible = "simple-audio-card";
+		simple-audio-card,format = "i2s";
+		simple-audio-card,name = "rockchip,rt5640-codec";
+		simple-audio-card,mclk-fs = <256>;
+		simple-audio-card,widgets =
+			"Microphone", "Mic Jack",
+			"Headphone", "Headphone Jack";
+		simple-audio-card,routing =
+			"DMIC1", "Mic Jack",
+			"DMIC2", "Mic Jack",
+			"Mic Jack", "MICBIAS1",
+			"IN2P", "Mic Jack",
+			"Headphone Jack", "HPOL",
+			"Headphone Jack", "HPOR";
+		simple-audio-card,cpu {
+			sound-dai = <&i2s_8ch>;
+		};
+		simple-audio-card,codec {
+			sound-dai = <&rt5640>;
+		};
+	};
+
 	backlight: backlight {
 		compatible = "pwm-backlight";
 		pwms = <&pwm0 0 25000 PWM_POLARITY_INVERTED>;
@@ -517,6 +540,17 @@
 
 };
 
+&i2c1 {
+	status = "okay";
+
+	rt5640: rt5640@1c {
+		#sound-dai-cells = <0>;
+		compatible = "realtek,rt5640";
+		reg = <0x1c>;
+		status = "disabled";
+	};
+};
+
 &i2c2 {
 	status = "okay";
 
@@ -540,6 +574,14 @@
 	};
 };
 
+&i2s_8ch {
+	status = "okay";
+	rockchip,i2s-broken-burst-len;
+	rockchip,playback-channels = <8>;
+	rockchip,capture-channels = <2>;
+	#sound-dai-cells = <0>;
+};
+
 &pwm0 {
 	status = "okay";
 };
-- 
2.35.3

